test/gateway/ost_prime/wrap.js

```diff
@@ -0,0 +1,209 @@
+const OSTPrime = artifacts.require("TestOSTPrime")
+  , BN = require('bn.js');
+
+const Utils = require('../../../test/test_lib/utils');
+const EventDecoder = require('../../test_lib/event_decoder.js');
+
+contract('OSTPrime.sol', function (accounts) {
+
+  describe('Unwrap', async () => {
+
+    const DECIMAL = new BN(10);
+    const POW = new BN(18);
+    const DECIMAL_FACTOR = DECIMAL.pow(POW);
+    const TOKENS_MAX = new BN(800000000).mul(DECIMAL_FACTOR);
+
+    let brandedTokenAddress, ostPrime, callerAddress, amount;
+
+    async function initialize(){
+      await ostPrime.initialize(
+        {from: accounts[5], value:TOKENS_MAX}
+      );
+    };
+
+    beforeEach(async function () {
+
+      brandedTokenAddress = accounts[2];
+      ostPrime = await OSTPrime.new(brandedTokenAddress);
+
+      callerAddress = accounts[3];
+      amount = new BN(500);
+
+      await ostPrime.setTokenBalance(ostPrime.address, amount);
+
+    });
+
+    it('Should fail when the payable amount is zero', async function () {
+
+      await initialize();
+
+      let amount = 0;
+      await Utils.expectRevert(
+        ostPrime.wrap.call({from: callerAddress, value: amount }),
+        "Payable amount should not be zero."
+      );
+
+    });
+
+    it('Should fail when the contract is not initialized', async function () {
+
+      await Utils.expectRevert(
+        ostPrime.wrap.call({from: callerAddress, value: amount }),
+        "Contract is not initialized."
+      );
+
+    });
+
+    it('Should fail when the payable amount sent is less than the available ' +
+      'balance', async function () {
+
+      await initialize();
+
+      // @dev this test with .call does not revert as expected.
```

Comment:
Because `.call` does not send any value and the balance is still available on the account, right? Also, at this point in time the balance of `accounts[9]` could be anything and should be used to deduct which amount should be used.

Also, like mentioned in the `OSTPrime` contract, I think a transfer of more than 50% of an account's balance would fail even though it should pass.